pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/unwrapped.nix

```diff
@@ -0,0 +1,150 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  callPackage,
+  pkg-config,
+  cmake,
+  ninja,
+  clang,
+  python3,
+  tg_owt ? callPackage ./tg_owt.nix { inherit stdenv; },
+  qtbase,
+  qtimageformats,
+  qtsvg,
+  qtwayland,
+  kcoreaddons,
+  lz4,
+  xxHash,
+  ffmpeg,
+  protobuf,
+  openalSoft,
+  minizip,
+  libopus,
+  alsa-lib,
+  libpulseaudio,
+  range-v3,
+  tl-expected,
+  hunspell,
+  gobject-introspection,
+  jemalloc,
+  rnnoise,
+  microsoft-gsl,
+  boost,
+  ada,
+  libicns,
+  apple-sdk_15,
+  nix-update-script,
+}:
+
+# Main reference:
+# - This package was originally based on the Arch package but all patches are now upstreamed:
+#   https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/telegram-desktop
+# Other references that could be useful:
+# - https://git.alpinelinux.org/aports/tree/testing/telegram-desktop/APKBUILD
+# - https://github.com/void-linux/void-packages/blob/master/srcpkgs/telegram-desktop/template
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "telegram-desktop-unwrapped";
+  version = "5.6.3";
+
+  src = fetchFromGitHub {
+    owner = "telegramdesktop";
+    repo = "tdesktop";
+    rev = "v${finalAttrs.version}";
+    fetchSubmodules = true;
+    hash = "sha256-frz425V5eRulNVxCf457TWQAzU/f9/szD/sx3/LYQ2Y=";
+  };
+
+  postPatch = lib.optionalString stdenv.hostPlatform.isLinux ''
+    substituteInPlace Telegram/ThirdParty/libtgvoip/os/linux/AudioInputALSA.cpp \
+      --replace-fail '"libasound.so.2"' '"${lib.getLib alsa-lib}/lib/libasound.so.2"'
+    substituteInPlace Telegram/ThirdParty/libtgvoip/os/linux/AudioOutputALSA.cpp \
+      --replace-fail '"libasound.so.2"' '"${lib.getLib alsa-lib}/lib/libasound.so.2"'
+    substituteInPlace Telegram/ThirdParty/libtgvoip/os/linux/AudioPulse.cpp \
+      --replace-fail '"libpulse.so.0"' '"${lib.getLib libpulseaudio}/lib/libpulse.so.0"'
+  '';
+
+  nativeBuildInputs =
+    [
+      pkg-config
+      cmake
+      ninja
+      python3
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      # to build bundled libdispatch
+      clang
+      gobject-introspection
+    ];
+
+  buildInputs =
+    [
+      qtbase
+      qtimageformats
+      qtsvg
+      lz4
+      xxHash
+      ffmpeg
+      openalSoft
+      minizip
+      libopus
+      range-v3
+      tl-expected
+      rnnoise
+      tg_owt
+      microsoft-gsl
+      boost
+      ada
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      protobuf
+      qtwayland
+      kcoreaddons
+      alsa-lib
+      libpulseaudio
+      hunspell
+      jemalloc
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      apple-sdk_15
+      libicns
+    ];
+
+  dontWrapQtApps = true;
```

Comment:
This does nothing as wrapQtAppsHook is not in nativeBuildInputs

Reply:
If it's not set qtbase's setup hook would complain.

Reply:
ah, ok